src/cli-client/internals/cli/cli_upload.go

```diff
@@ -0,0 +1,121 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/canonical/oci-factory/cli-client/internals/client"
+	"github.com/canonical/oci-factory/cli-client/internals/logger"
+	"github.com/canonical/oci-factory/cli-client/internals/trigger"
+)
+
+type UploadRelease struct {
+	Track     string
+	Risks     []string
+	EndOfLife string
+}
+
+type CmdUpload struct {
+	UploadRelease []string `long:"release" description:"Release images to container registries.\nSyntax: --release tracks=<release track>,risks=<risk1>[,<risk2>...],eol=yyyy-mm-dd"`
+}
+
+var releaseKeys = []string{"tracks", "risks", "eol"}
+
+func init() {
+	parser.Command.AddCommand("upload", "Trigger the build and release for the image in the current working directory",
+		"long description", &CmdUpload{})
+}
+
+func (c *CmdUpload) Execute(args []string) error {
+	releases, err := parseUploadReleases(c.UploadRelease)
+	if err != nil {
+		parser.WriteHelp(os.Stdout)
+		return err
+	}
+	triggerUploadReleases(releases)
+	return nil
+}
+
+// parseUploadReleases parses the release arguments and returns a list of UploadRelease structs.
+// The release arguments are expected to be in the format "tracks=<track>,risks=<risk1>[,<risk2>...],eol=yyyy-mm-dd".
+// Multiple release arguments can be passed, separated by spaces.
+func parseUploadReleases(args []string) ([]UploadRelease, error) {
+	var release UploadRelease
+	releases := make([]UploadRelease, 0)
+	regex := regexp.MustCompile("(((tracks=[^,]+)|(risks=[^,]+,*[^,]+)|(eol=[^,]+)),?){3}")
```

Comment:
We are not currently enforcing the ordering of the keys from the user input. @cjdcordeiro Shall we enforce that? If this is enforced, then using regex to extract the inputs would become much easier.

Reply:
Or maybe, I can match only a key-value pair of the string each time, using `tracks=([^,]+),?` to match tracks, `risks=(((beta|candidate|edge|stable),?)+),?` to match risks and `eol=(\d{4}-\d{2}-\d{2}),?` to match eol separately?

This could mitigate the complexity of implementing such a regex, and also provide finer-grained error messages for the user.

Reply:
Updated in [0c81222](https://github.com/canonical/oci-factory/pull/198/commits/0c81222bcd16b62d8d1ba393a0651fd3e5e584a9).